tests/test_dse.py

```diff
@@ -1172,16 +1168,18 @@ def test_from_different_nests(self, rotate):
                                                'cire-rotate': rotate}))
 
         # Check code generation
-        assert 'bf0' in op1._func_table
-        assert 'bf1' in op1._func_table
-        trees = retrieve_iteration_tree(op1._func_table['bf0'].root)
-        assert len(trees) == 2
-        assert trees[0][-1].nodes[0].body[0].write.is_Array
-        assert trees[1][-1].nodes[0].body[0].write is u
-        trees = retrieve_iteration_tree(op1._func_table['bf1'].root)
-        assert len(trees) == 2
-        assert trees[0][-1].nodes[0].body[0].write.is_Array
-        assert trees[1][-1].nodes[0].body[0].write is v
+        trees = retrieve_iteration_tree(op1)
+        assert len(trees) == 6
+
+        exprs = FindNodes(Expression).visit(op1)
```

Comment:
this looks much weaker than it is in master

Reply:
actually, I'm not sure what these extra checks on `exprs` would add ? AFAICT, the edits above make the test same as master?

Reply:
well.. why not.. I can delete if you want

Reply:
why not --> because they might uselessly increase verbosity (and thus complexity)
why yes --> they really have value

which one is it?